docs/instruments/ZJSONInstrument_v0.0.1.md

```diff
@@ -0,0 +1,1028 @@
+        It represents the helper's data type i.e. static, section, table, etc.
+
+`description`: Required string.
+               The human readable description of this helper, provided in every available language,
+               and also the text to be displayed in the instrument.
+
+It may be rendered on the front-end as a header, depending on the helper type i.e. this will be the
+content of the static helper itself, the header of the section helper, or column header for the table.
+
+`options`: Required object of type-specific keys.
+           It contains the type-dependent options for this helper, further described in section
+           [4.0: Helper Types](#4.0:-helper-types).
+
+`rules`: Required object with keys whose value may be an empty string as described for `fields`.
+While instrument `fields` have corresponding `data` associated with it upon form submission, a `helper` is rendered on the front-end as an integral element that however does not collect data. There are
+5 types of helpers: the static element, and 4 other layout-specific formatting elements, group, row, table, and section.
+
+## 4.1: Helper Types
+
+### 4.1.1: Static
+
+A `helper` of type "static" represents headings, sub-headings, labels, or informational text that is required in
+an instrument. These can be rendered on the front-end as a header, static, or link element.
+Static helpers are some text displayed to provide information and without an accompanying data input.
+Static helpers do not have an `options` key.
+
+```js
+{
+    "schema" : {
+        ...
+        "helpers" : {
+            "nameOfHelper" : {
+                "type": "static",
+                "description": {
+                    ...
+                },
+                "rules": {
+                    ...
+                }
+        ...
+}
+```
+
+### 4.1.2: Group
+
+A helper of type "group" denotes a horizontal collection of elements which should be displayed
+together and is separated by a delimiter.
+
+They often have rules which work together and may be interdependent. Groups have the following form:
+
+```js
+{
+    "schema" : {
+        ...
+        "helpers" : {
+            "nameOfHelper" : {
+                "type": "group",
+                "description": {
+                    ...
+                },
+                "options": {
+                    "order": [
+                        string,
+                        string,
+                        ...
+                    ]
+                },
+                "rules": {
+                    ...
+                }
+        ...
+}
+```
+
+`options.order`: Required array of strings.
+                 An array of the names of `fields` or `helpers`, in order, that are to be formatted
+                 within this group. It may contain any field, but only helpers of the type "static".
+ 
+### 4.1.3: Row
+
+A helper of type "row" denotes a row in a table. If rows are not in a table, they should be rendered
+as a "group" helper.
```

Comment:
In that case, why have "group" helpers and not just define groups as rows that aren't embedded in a table? (or vice versa)?

Reply:
I might be able to answer that one. Due to a recent bug we found in CCNA I think its a good idea to distinguish rows which should depend on the headers of the table they belong in  and groups which are just floating elements grouped together.

the bug: because in the current implementation you define a group of headers and add them to the page and then subsequently define groups of input fields (rows in the table) and add them to the page some  `n` number of times. During a routine bugfix, 2 elements of the hearder (assume header2 and header 3) group were switched (assume header2 went to column 3 and vice versa) and that was not caught during code review, which means that since that was released users had been entering values for header 3 in column 2 and values for header 2 in column 3..... naturally this was a mess to clean up after and we thought it would be a good idea to make tables more strict (ie not adding rows as independent groups). If i remember correctly the idea was to define the header of the column of a field in each field of a row and thus in case the headers dont line up, an error would be thrown

Reply:
see line 772

Reply:
There's also an argument to be made that separating them are more intuitive for people to understand, I was just wondering if any thought had went into combining them since it would simplify implementations.